### PR TITLE
[filters] Use Eigen instead of handwritten logic in `SamplingSurfaceNormal::findXYZMaxMin`

### PR DESCRIPTION
**Pull request by kunaltyagi**
Follow-up for https://github.com/PointCloudLibrary/pcl/pull/4190#discussion_r446896672

### CODE REVIEW

File: filters/include/pcl/filters/impl/sampling_surface_normal.hpp [hunk 1]
```diff
@@ -66,56 +66,19 @@ pcl::SamplingSurfaceNormal<PointT>::applyFilter (PointCloud &output)
...
-    }
-    if (cloud[i].y < minval)
-    {
-      minval = cloud[i].y;
-    }
+  // 4f to ease vectorization
+  Eigen::Array4f min_array =
+      Eigen::Vector4f::Ones() * std::numeric_limits<float>::max();
+  Eigen::Array4f max_array =
+      Eigen::Vector4f::Ones() * std::numeric_limits<float>::lowest();
```
Comment (SergioRAgostinho) [inline filters/include/pcl/filters/impl/sampling_surface_normal.hpp:73]: `Eigen::Array4f::Constant()` expresses better what you're trying to do here.

**Comment by kunaltyagi** [comment]
Eigen appraoch is more general and will be allowed for unaligned types as well. Will that explode?

**Comment by SergioRAgostinho** [comment]
I'm sorry, I don't understand the question.

**Comment by kunaltyagi** [comment]
The elements in Eigen array are aligned, (ensured by the `getXMap()`). If we use Eigen on raw data, we need to ensure alignment ourselves, right?

**Comment by SergioRAgostinho** [comment]
Probably, however I've never test it.

**Comment by kunaltyagi** [comment]
Neither have I, but it's just a fear I have since Eigen explicitly requires aligned data. If possible, I'd request to investigate possible Eigen-ization of loops later and separately.

**Comment by SergioRAgostinho** [comment]
Sure.

**Pull request merged**